server/src/main/java/org/elasticsearch/http/HttpStats.java

```diff
@@ -90,7 +108,24 @@ public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerP
                     .startArray(Fields.CLIENTS)
             ),
             clientStats.iterator(),
-            Iterators.single((builder, params) -> builder.endArray().endObject())
+            Iterators.single((builder, params) -> {
+                builder.endArray();
+                if (httpRouteStats.isEmpty() == false) {
```

Comment:
Let's just always output this field even if empty, it makes life harder for the clients if fields come and go.

Reply:
Yep removed the if check.